x-pack/plugins/data_enhanced/config.ts

```diff
@@ -23,6 +23,13 @@ export const configSchema = schema.object({
        * trackingInterval controls how often we track search session objects progress
        */
       trackingInterval: schema.duration({ defaultValue: '10s' }),
+
+      /**
+       * trackingTimeout controls for how long task manager waits for search session monitoring task to complete before considering it timed out,
+       * If tasks timeouts it receives cancel signal and next task starts in "trackingInterval" time
+       */
+      trackingTimeout: schema.duration({ defaultValue: '5m' }),
```

Comment:
'5m' is the current default in task manager. I don't change the default and expose a way to override it.